### DISCUSSION TITLE
Is it possible to have different traits value of component for different devices?

### DISCUSSION THREAD

**Discussion by pittleCheung** [GrapesJS/grapesjs#5502] [Ideas]
Is it possible to make traits like style manager, inherit the configuration of the previous device, and have separate traits for each device?
<img width="1674" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/114127651/9e6f9de0-8a5b-45a0-a73b-ee664a23b4e8">
<img width="1669" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/114127651/66aea262-0cf7-4ba7-8890-239dfa433891">

Is it possible to set device for each device?
<img width="1674" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/114127651/52cdd013-0cfc-45f9-a6ab-debeefc69617">
<img width="1668" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/114127651/c192830a-0cee-4db2-9c0e-8b13a39f5e72">

my inspiration comes from https://github.com/GrapesJS/grapesjs/issues/2192

**Comment by pittleCheung** [comment]
Is it possible to set trait value for each device?

**Comment by glaprida** [comment]
I don't think so, since traits render as element attributes, and those aren't affected by css media queries.

You could add 3 different 'position' traits (one for each device) and show/hide them with some custom code depending on the current device. A bit hacky, I know, but I think it should work.

**Comment by artf** [comment]
> I don't think so, since traits render as element attributes, and those aren't affected by css media queries.

👆
